types/node_definition.go

```diff
@@ -132,6 +133,13 @@ func (n *NodeDefinition) GetImage() string {
 	return n.Image
 }
 
+func (n *NodeDefinition) GetImagePullPolicy() string {
+	if n == nil {
+		return ""
+	}
+	return n.ImagePullPolicy
+}
+
```

Comment:
@steiler in [fe6d9fd](https://github.com/srl-labs/containerlab/pull/1223/commits/fe6d9fd0dbfb259c5ce0869c4aed55e9b4d66c37) I have changed this piece to return a basic string type, since I think *NodeDefinition methods such as this one, should return the raw values users provided in the topology file, without implying defaults.

Defaults, are handled in the Topology.GetNodeImagePullPolicy instead, where we typically do that.

Check if that makes sense to you as well.

Reply:
yes, looks alright!